server/mailet/integration-testing/src/test/java/org/apache/james/mailets/RemoteDeliveryErrorTest.java

```diff
@@ -68,64 +73,44 @@
-        "}]";
-    private static final String SINGLE_500_RCPT_BEHAVIOR = "[{" +
-        "  \"condition\": {\"operator\":\"matchAll\"}," +
-        "  \"response\": {\"code\":521, \"message\":\"mock response\", \"rejected\":true}," +
-        "  \"command\": \"RCPT TO\"," +
-        "  \"numberOfAnswer\": 1" +
-        "}]";
-    private static final String SINGLE_500_FROM_BEHAVIOR = "[{" +
-        "  \"condition\": {\"operator\":\"matchAll\"}," +
-        "  \"response\": {\"code\":521, \"message\":\"mock response\", \"rejected\":true}," +
-        "  \"command\": \"MAIL FROM\"," +
-        "  \"numberOfAnswer\": 1" +
-        "}]";
-    private static final String SINGLE_500_DATA_BEHAVIOR = "[{" +
-        "  \"condition\": {\"operator\":\"matchAll\"}," +
-        "  \"response\": {\"code\":521, \"message\":\"mock response\", \"rejected\":true}," +
-        "  \"command\": \"DATA\"," +
-        "  \"numberOfAnswer\": 1" +
-        "}]";
+
+    private static final MockSMTPBehavior.NumberOfAnswersPolicy TWO_TIMES = MockSMTPBehavior.NumberOfAnswersPolicy.times(2);
+    private static final MockSmtpBehaviors ALWAYS_421_RCPT_BEHAVIOR = reject421Behavior(SMTPCommand.RCPT_TO);
+    private static final MockSmtpBehaviors ALWAYS_421_FROM_BEHAVIOR = reject421Behavior(SMTPCommand.MAIL_FROM);
+    private static final MockSmtpBehaviors ALWAYS_421_DATA_BEHAVIOR = reject421Behavior(SMTPCommand.DATA);
+    private static final MockSmtpBehaviors TWICE_421_RCPT_BEHAVIOR = reject421Behavior(SMTPCommand.RCPT_TO, TWO_TIMES);
+    private static final MockSmtpBehaviors TWICE_421_FROM_BEHAVIOR = reject421Behavior(SMTPCommand.RCPT_TO, TWO_TIMES);
+    private static final MockSmtpBehaviors TWICE_421_DATA_BEHAVIOR = reject421Behavior(SMTPCommand.RCPT_TO, TWO_TIMES);
+    private static final MockSmtpBehaviors SINGLE_500_RCPT_BEHAVIOR = reject521Behavior(SMTPCommand.RCPT_TO);
+    private static final MockSmtpBehaviors SINGLE_500_FROM_BEHAVIOR = reject521Behavior(SMTPCommand.MAIL_FROM);
+    private static final MockSmtpBehaviors SINGLE_500_DATA_BEHAVIOR = reject521Behavior(SMTPCommand.DATA);
     private static final String BOUNCE_MESSAGE = "Hi. This is the James mail server at localhost.\n" +
         "I'm afraid I wasn't able to deliver your message to the following addresses.\n" +
         "This is a permanent error; I've given up. Sorry it didn't work out. Below\n" +
         "I include the list of recipients and the reason why I was unable to deliver\n" +
         "your message.";
     private static final ResponseSpecification RESPONSE_SPECIFICATION = new ResponseSpecBuilder().build();
 
+    private static MockSmtpBehaviors reject421Behavior(SMTPCommand smtpCommand) {
+        return reject421Behavior(smtpCommand, MockSMTPBehavior.NumberOfAnswersPolicy.anytime());
+    }
+
+    private static MockSmtpBehaviors reject421Behavior(SMTPCommand smtpCommand, MockSMTPBehavior.NumberOfAnswersPolicy numberOfAnswer) {
+        return new MockSmtpBehaviors(
```

Comment:
what about having a fancy builder that would render all these methods useless?

Reply:
Agree, but can we do that as an enhancement in a followup PR?

Reply:
Finaly I contributed this here.